mitmproxy/contentviews/base.py

```diff
@@ -34,6 +33,18 @@ def __call__(self, data: bytes, **metadata) -> TViewResult:
         """
         raise NotImplementedError()  # pragma: no cover
 
+    def should_render(self, content_type) -> bool:
```

Comment:
While we are improving the API here, how would you all feel about having this function return an `int` as priority, where a value > 0 signals that the view can render this?

We could then selected the preferred view like this:

```python
priority, view = max(
    (v.render_priority(ct), v)
    for v in contentviews.views
)
```

A major benefit of this approach would be that I can write a custom contentview addon that pretty-prints data in a specific way for me. For example, let's say I'm only interested in a specific subproperty of a 2MB JSON blob, I could write an addon with a higher priority that only displays that part. I would have found that useful a couple of times in the past.

Reply:
Great idea. But this would require changing `should_render` interface even further – passing the request/response contents as an argument so the extension could check if the JSON subproperty exists. I am afraid that this is going to grow the PR.

On the other hand, if we implemented your proposition, we could minimise the number of incompatible changes for 3-rd party views.

Do you think we should discuss the `should_render` interface now, or should we skip it for now?